GiniVision/Classes/Core/Screens/Camera/Camera.swift

```diff
@@ -224,23 +228,27 @@ fileprivate extension Camera {
     
     func setupInput() {
         // Specify that we are capturing a photo, this will reset the format to be 4:3
-        self.session.sessionPreset = .photo
-        if self.session.canAddInput(self.videoDeviceInput!) {
-            self.session.addInput(self.videoDeviceInput!)
-        } else {
-            Log(message: "Could not add video device input to the session", event: .error)
+        session.sessionPreset = .photo
+        if let input = videoDeviceInput {
+            if !session.canAddInput(input) {
+                for previousInput in session.inputs {
+                    session.removeInput(previousInput)
+                }
+            }
+            session.addInput(input)
         }
     }
     
     func setupPhotoCaptureOutput() {
         let output = AVCapturePhotoOutput()
-        
-        if self.session.canAddOutput(output) {
-            self.session.addOutput(output)
-            self.photoOutput = output
-        } else {
-            Log(message: "Could not add still image output to the session", event: .error)
+
+        if !session.canAddOutput(output) {
```

Comment:
This is duplicated. Maybe we should put the whole block in a method like `ensureCanAddOutputToSession()`?

Reply:
There are 2 different types of output:
AVCapturePhotoOutput for photo and
AVCaptureMetadataOutput for qr

Reply:
OK, nevermind.